edge-apps/powerbi/static/js/main.js

```diff
@@ -1,95 +1,118 @@
 /* global screenly, panic */
 
-(function () {
-  const DEFAULT_TOKEN_REFRESH_SEC = 30 * 60; // refresh token every 30 minutes
+;(function () {
+  const MIN_TOKEN_REFRESH_MIN = 1 // minimum 1 minute
+  const DEFAULT_TOKEN_REFRESH_MIN = 30 // default 30 minutes
```

Comment:
The setting should call refresh(), but not request a new embed token.

Actually, we have to change the logic, we need to store the embedded token somewhere(localStorage for example) and take it instead of requesting a new one if it hasn't expired. Because Power BI has the API requests limitation.

Reply:
@rusko124 Updated the token refresh logic. 

- Store embed tokens in localStorage with expiry timestamps (1 hour)
- Reuse tokens if valid (more than 5 minutes until expiry)
- Added configurable refresh interval (default: 30 minutes, minimum: 1 minute)
- Ensures tokens are checked at least as often as the app refreshes
- Automatically clear invalid tokens from cache

Please check. Thanks.